### PR TITLE
added nodes/types endpoint with testcase

### LINKED ISSUE #3266: Add /nodes/types endpoint for develop branch
Implement /nodes/types endpoint to return unique list of all available node types stored in database.

### PR DESCRIPTION
**Pull request by waychal**
It returns unique list of all the nodes types available in database.
Fixes #3266

This endpoint will be updated again when we will rename all the existing endpoints according to rest api version 3 standards [#3181].

### CODE REVIEW

**Comment by CasperWA** [comment]
> This endpoint will be updated again when we will rename all the existing endpoints according to rest api version 3 standards [#3181].

Does this mean the PR cannot be merged until #3181 is merged, or simply that another PR will have to be made?

Also, you seem to have some pre-commit issues according to [Travis](https://travis-ci.org/aiidateam/aiida-core/jobs/574798792) :)

**Comment by waychal** [comment]
> 
> Does this mean the PR cannot be merged until #3181 is merged, or simply that another PR will have to be made?

This PR needs to be merged now into develop branch. Later PR #3181 will be updated to adapt the changes from this PR.
> 
> Also, you seem to have some pre-commit issues according to [Travis](https://travis-ci.org/aiidateam/aiida-core/jobs/574798792) :)

Thanks. I have fixed it now and tests are running again.

**Comment by CasperWA** [comment]
> This PR needs to be merged now into develop branch. Later PR #3181 will be updated to adapt the changes from this PR.

Great! Thanks for the clarification :)

Still some [troubles](https://travis-ci.org/aiidateam/aiida-core/jobs/575236404#L1349) with prospector for the pre-commit it seems?
I can recommend installing the pre-commit git hook for your repo. To get the proper modules installed you can use the `[dev_precommit]` when installing AiiDA, like`pip install -U -e .[dev_precommit]` and then, in the repo top-level also, write `pre-commit install` :)

**Comment by waychal** [comment]
I have pre-commit installed on my machine and all hooks are passing there. Tests are still running for this PR. So lets check again.

**Comment by CasperWA** [comment]
> I have pre-commit installed on my machine and all hooks are passing there. Tests are still running for this PR. So lets check again.

Ah - sorry. Seb updated the hooks recently though, so you may need to update them? :)

File: aiida/restapi/api.py [hunk 1]
```diff
@@ -196,6 +199,7 @@ def __init__(self, app=None, **kwargs):
         self.add_resource(
             Code,
             '/codes/',
+            '/codes/types/',
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/restapi/api.py]: I agree. I have made the changes and updated the branch.

**Comment by sphuber** [comment]
This PR has not gone quite as it should. During the merging/rebasing various commits from other PRs were included that have nothing to do with this PR. Luckily, during the merge squash only the correct changes were included, but as you can see, the commit message still contains the messages of the unintended commits. I have to say that I find this a bit sloppy. Mistakes can of course happen, but when you merge, please make sure that everything is correct. Just reading through the commit message is the minimum that should be done

**Pull request merged**